File: src/main/java/io/github/saneea/feature/time/format/DateTimeFormatterFormat.java
package io.github.saneea.feature.time.format;

import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;

public class DateTimeFormatterFormat implements Format {

	private final DateTimeFormatter formatter;

	public DateTimeFormatterFormat(DateTimeFormatter formatter) {
		this.formatter = formatter;
	}

	@Override
	public ZonedDateTime parse(String time) {
		return ZonedDateTime.from(formatter.parse(time));
	}

	@Override
	public String render(ZonedDateTime time) {
		return formatter.format(time);
	}

}

File: src/main/java/io/github/saneea/dvh/feature/time/format/DateTimeFormatterFormat.kt
package io.github.saneea.dvh.feature.time.format

import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter

class DateTimeFormatterFormat(private val formatter: DateTimeFormatter) : Format {

    override fun parse(time: String) = ZonedDateTime.from(formatter.parse(time))!!

    override fun render(time: ZonedDateTime) = formatter.format(time)!!
}